### DISCUSSION TITLE
Etsy Authentication flow passing same state while requesting an Authorization Code.

### DISCUSSION THREAD

**Discussion by nikhild-tech** [etsy/open-api#1033] [Q&A]
Hi,

In my current authentication flow when requesting an Authorization Code, I have been using the same state parameter called "superstate," which is not unique for every user. Over the past 20-25 days, I have noticed a sudden increase in the occurrence of two errors:
1) refresh_token is invalid
2) refresh_token is revoked
Could you please clarify whether these errors are caused by my failure to pass a unique state parameter when requesting an Authorization Code, or if there might be another reason for their occurrence?

**Comment by blipchin26** [comment]
The problem is likely the fact that you do not have a unique state code. If you're in Python you can generate a unique state like so:

import secrets

state = secrets.token_hex(16)

This will generate a 16 character hexadecimal code which will be totally unique for each request. I haven't received a single refresh_token error since using it so I hope this helps!

**Comment by emergingdzns** [comment]
Also, it's entirely possible that some of the users that has added your application later decided they didn't need it and removed it from their shop. This would automatically trigger a revocation of the access token and refresh token. Also, if a user changes their password, Etsy assumes (out of abundance of caution) that their account may have been compromised so they revoke all tokens upon the password change to make sure no one has "backdoor" access to the account.
